<launch>
    <arg name="name" />
    <arg name="domain" />

    <arg name="map_frame" default="map" />
    <arg name="robot_frame" default="base_link" />

    <arg name="use_sim_time" default="false" />

    <node pkg="pose_publisher" exec="pub_node" name="pose_pub_$(var name)">
        <param name="map_frame" value="$(var map_frame)" />
        <param name="robot_frame" value="$(var robot_frame)" />
        <param name="out_map_frame" value="$(var map_frame)" />
        <param name="out_robot_frame" value="$(var name)" />
        <param name="use_sim_time" value="$(var use_sim_time)" />

        <remap from="pose" to="/robot_poses" />
    </node>

    <node pkg="central_nav" exec="path_pub_node" name="path_pub_$(var name)">
        <param name="robot_name" value="$(var name)" />
        <remap from="path_in" to="/plan" />
        <remap from="path_out" to="/robot_paths" />
    </node>

    <include file="$(find-pkg-share domain_bridge)/launch/domain_bridge.launch.xml">
        <arg name="config" value="$(find-pkg-share central_nav)/config/bridge_config.yml"/>
        <arg name="from_domain" value="$(env ROS_DOMAIN_ID 0)" />
        <arg name="to_domain" value="$(var domain)" />
    </include>
</launch>